reascripts/ReaSpeech/source/ReaSpeechWelcomeUI.lua

```diff
@@ -0,0 +1,159 @@
+--[[
+
+ReaSpeechWelcomeUI.lua - Welcome screen for ReaSpeech
+
+]]--
+
+ReaSpeechWelcomeUI = Polo {
+  TITLE = "Welcome!",
+
+  WIDTH = 610,
+  HEIGHT = 800,
+
+  HEADING_MARGIN = 8,
+  PADDING = 12,
+  PARA_MARGIN = 2,
+
+  FOOTER_BG_COLOR = 0x222222ff,
+  FOOTER_HEIGHT = 40,
+  FOOTER_MARGIN = 5,
+
+  HOME_URL = "https://techaud.io/reaspeech/",
+  GITHUB_URL = "https://github.com/TeamAudio/reaspeech",
+  DOCKER_HUB_URL = "https://hub.docker.com/r/techaudiodoc/reaspeech",
+}
+
+function ReaSpeechWelcomeUI:init()
+  self.is_demo = self.is_demo or false
+  self.is_open = false
+  self.presenting = false
+
+  self.url_opener_cmd = '/usr/bin/open "%s"'
+  if reaper.GetOS():match('Win') then
+    self.url_opener_cmd = 'start "" "%s"'
+  end
+end
+
+function ReaSpeechWelcomeUI:present()
+  self.presenting = true
+end
+
+function ReaSpeechWelcomeUI:render()
+  if not self.presenting then
+    return
+  end
+
+  local opening = not self.is_open
+  if opening then
+    self:_open()
+  end
+
+  local center = {ImGui.Viewport_GetCenter(ImGui.GetWindowViewport(ctx))}
+  ImGui.SetNextWindowPos(ctx, center[1], center[2], ImGui.Cond_Appearing(), 0.5, 0.5)
+  ImGui.SetNextWindowSize(ctx, self.WIDTH, self.HEIGHT, ImGui.Cond_FirstUseEver())
+
+  ImGui.PushStyleVar(ctx, ImGui.StyleVar_WindowPadding(), 0, 0)
+  app:trap(function ()
+    if ImGui.BeginPopupModal(ctx, self.TITLE, true, ImGui.WindowFlags_AlwaysAutoResize()) then
+      app:trap(function ()
+        self:render_content()
+      end)
+      ImGui.EndPopup(ctx)
+    else
+      self:_close()
+    end
+  end)
+  ImGui.PopStyleVar(ctx)
+end
+
+function ReaSpeechWelcomeUI:render_content()
+  self:render_banner()
+  self:render_welcome_text()
+  if self.is_demo then
+    self:render_demo_text()
+  end
+  self:render_close_button()
+  self:render_footer()
+end
+
+function ReaSpeechWelcomeUI:render_banner()
+  app.png_from_bytes('reaspeech-banner')
+end
+
+function ReaSpeechWelcomeUI:render_welcome_text()
+  self:render_heading("Welcome to ReaSpeech!")
+  self:render_text([[This is a tool for transcribing audio in REAPER.]])
+end
+
+function ReaSpeechWelcomeUI:render_demo_text()
+  self:render_heading("Demo Version")
+  self:render_text([[Please note that this version is a demo and may not be available at all times.
+For a more reliable experience, you can run ReaSpeech locally using the Docker image.]])
+end
```

Comment:
maybe "Docker image" can link to some docs or our hub page?

Reply:
Done, thanks!